Tests/PINRemoteImageTests.m

```diff
@@ -392,12 +398,14 @@ - (void)testImageRequestIgnoresLocalData
     
     self.imageManager = [[PINRemoteImageManager alloc] initWithSessionConfiguration:nil];
     self.imageManager.sessionManager.delegate = self;
-    
-    [self.imageManager setRequestConfiguration:^NSURLRequest * _Nonnull(NSURLRequest * _Nonnull request) {
+
+    // To help reduce possibilities of flakiness, assign directly
+    // instead of `setRequestConfiguration` to avoid locks
+    self.imageManager.requestConfigurationHandler = ^NSURLRequest * _Nonnull(NSURLRequest * _Nonnull request) {
         XCTAssertEqual(request.cachePolicy, NSURLRequestReloadIgnoringLocalCacheData);
         [expectation fulfill];
         return request;
-    }];
+    };
   
```

Comment:
I think if we avoid using the locks we may create flakiness in a different way :( Does calling `[self waitForExpectationsWithTimeout:[self timeoutTimeInterval] handler:nil];` after calling setRequestConfiguration: solve the flakiness?

Reply:
Oh man, I'm so sorry this is designed this way 😅 

I think we need to add a new API:
`setRequestConfiguration:completion` (which the old `setRequestConfiguration:` can call with a nil completion). Then we can call the new API in the test method with a new expectation that is fulfilled in the completion block. We'll then wait for just that expectation to be fulfilled before calling `downloadImageWithURL:…`